core/data/dictionary/dictionary.go

```diff
@@ -33,7 +33,7 @@ type I interface {
 	Entries() []Entry
 	KeyTy() reflect.Type
 	ValTy() reflect.Type
-	KeysSorted() []interface{}
+	Keys() []interface{}
```

Comment:
maybe add a comment here, that the keys are expected to be sorted?

Reply:
They are not expected to be sorted. There's only expectations in the serialization, which is not part of this contract. However, you are right that there should be comments (which are coming in a later CL).